src/Sil/Component/Address/Model/CountryInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright (C) 2015-2018 Libre Informatique
+ *
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Component\Address\Model;
+
+interface CountryInterface
+{
+    /**
+     * Return the name of current country.
+     *
+     * @return string
+     */
+    public function getName(): string;
+
+    /**
+     * Gets country code.
+     *
+     * @return string
+     */
+    public function getCode(): string;
+
+    /**
+     * Sets the city code.
+     *
+     * @param string $code
+     */
+    public function setCode(string $code): void;
```

Comment:
Why setCode in the interface when code is already a mandatory parameter of the constructor